src/KubernetesClient.Informers/ResourceObservableExtensions.cs

```diff
@@ -0,0 +1,273 @@
+                        if (comparer != null)
+                        {
+                            var previouslyKnownEntitiesInResetWindowKeys = cacheSnapshot
+                                .Keys
+                                .Intersect(resetBuffer.Select(x => keySelector(x.Value)));
+
+                            updatedEntities = resetBuffer
+                                .Where(x => previouslyKnownEntitiesInResetWindowKeys.Contains(keySelector(x.Value)))
+                                .Select(x => x.Value) // stuff in buffer that also existed in cache (by key)
+                                .Except(cacheSnapshot.Select(x => x.Value), comparer)
+                                .Select(x => x.ToResourceEvent(EventTypeFlags.Modify | EventTypeFlags.Computed))
+                                .ToList();
+                        }
+
+                        var updatedKeys = updatedEntities
+                            .Select(x => keySelector(x.Value))
+                            .ToHashSet();
+
+                        var resetEntities = resetBuffer
+                            .Select(x => x.Value)
+                            .Where(x => x != null &&
+                                        !addedKeys.Contains(keySelector(x)) &&
+                                        !deletedKeys.Contains(keySelector(x)) &&
+                                        !updatedKeys.Contains(keySelector(x)))
+                            .ToReset()
+                            .ToList();
+
+                        return deletedEntities
+                            .Union(addedEntities)
+                            .Union(updatedEntities)
+                            .Union(resetEntities);
+                    })
+                    .SynchronizeCache(cache, keySelector)
+                    .Do(msg => { cacheSynchronized = true; })
+                    .Select(x => x.Value)
+                    .ObserveOn(Scheduler.Immediate)
+                    .Subscribe(observer);
+            });
+        }
+
+        /// <summary>
+        ///     Injects a <see cref="ResourceEvent{T}" /> of type <see cref="ResourceStreamType.Sync" /> into the observable for each item produced
+        ///     by the <see cref="ResourceStreamType.List" /> operation from <paramref name="source" />
+        /// </summary>
+        /// <param name="source">The source sequence that will have sync messages appended</param>
+        /// <param name="timeSpan">The timespan interval at which the messages should be produced</param>
+        /// <typeparam name="T">The type of resource</typeparam>
+        /// <returns>Original sequence with resync applied</returns>
+        public static IObservable<ResourceEvent<T>> Resync<T>(this IObservable<ResourceEvent<T>> source, TimeSpan timeSpan)
+        {
+            return Observable.Create<ResourceEvent<T>>(observer =>
+            {
+                var timerSubscription = Observable
+                    .Interval(timeSpan)
```

Comment:
This needs a Scheduler parameter, probably should be an optional parameter of the Resync method